pkg/deploy/reaper/reaper.go

```diff
@@ -46,8 +59,13 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, gracePeriod *
 	if err != nil {
 		return "", err
 	}
-	// Remove all the deployments of the configuration
-	for _, rc := range util.ConfigSelector(name, rcList.Items) {
+
+	// If there is neither a config nor any deployments, we can return NotFound.
+	deployments := util.ConfigSelector(name, rcList.Items)
+	if len(deployments) == 0 {
+		return "", kerrors.NewNotFound("DeploymentConfig", name)
```

Comment:
Fixed and added test coverage (which revealed lots of holes in the Fake stuff which we'll want to improve as we go; in this case, you couldn't simulate returning an error from a delete).